vault/core.go

```diff
@@ -1069,14 +1069,12 @@ func (c *Core) sealInitCommon(ctx context.Context, req *logical.Request) (retErr
 		RootPrivsRequired: true,
 	})
 	if !authResults.Allowed {
-		if authResults.DeniedError {
-			retErr = multierror.Append(retErr, logical.ErrPermissionDenied)
-		}
-		if authResults.Error.ErrorOrNil() != nil {
-			retErr = multierror.Append(retErr, authResults.Error)
+		if authResults.Error.ErrorOrNil() == nil || authResults.DeniedError {
+			c.stateLock.RUnlock()
```

Comment:
Since both of these return, maybe it makes sense to release the lock before both of these calls instead of putting it before both of these returns.

Reply:
Right again. Fixed.